src/ignored_suffixes.rs

```diff
@@ -0,0 +1,109 @@
+use std::ffi::OsStr;
+use std::path::Path;
+
+use crate::error::*;
+
+const IGNORED_SUFFIXES: [&str; 13] = [
+    // Editor etc backups
+    "~",
+    ".bak",
+    ".old",
+    ".orig",
+    // Debian and derivatives apt/dpkg/ucf backups
+    ".dpkg-dist",
+    ".dpkg-old",
+    ".ucf-dist",
+    ".ucf-new",
+    ".ucf-old",
+    // Red Hat and derivatives rpm backups
+    ".rpmnew",
+    ".rpmorig",
+    ".rpmsave",
+    // Build system input/template files
+    ".in",
+];
+
+#[derive(Debug, Clone, Default)]
+pub struct IgnoredSuffixes {
+    values: Vec<String>,
+}
+
+impl IgnoredSuffixes {
+    pub fn new(values: Vec<String>) -> Self {
+        IgnoredSuffixes { values }
+    }
+
+    pub fn strip_suffix<'a>(&self, file_name: &'a str) -> Option<&'a str> {
+        for suffix in self
+            .values
+            .iter()
+            .map(|v| v.as_str())
+            .chain(IGNORED_SUFFIXES.iter().copied())
```

Comment:
I think I would prefer all ignored suffixes to be in `values`. Both default ones and the ones added by clients. This would avoid a special case in the code, which is generally nice.